op-deployer/pkg/deployer/state/intent.go

```diff
@@ -32,35 +33,313 @@ func (d DeploymentStrategy) Check() error {
+//  2. no non-standard values for standard fields (user should not have changed these)
+func (c *Intent) validateStandardValues() error {
+	standardSuperchainRoles, err := getStandardSuperchainRoles(c.L1ChainID)
+	if err != nil {
+		return fmt.Errorf("error getting standard superchain roles: %w", err)
+	}
+	if *c.SuperchainRoles != *standardSuperchainRoles {
+		return fmt.Errorf("SuperchainRoles does not match standard value")
+	}
+
+	for _, chain := range c.Chains {
+		if chain.ID == emptyHash {
+			return fmt.Errorf("missing l2 chain ID")
+		}
+		if err := chain.Roles.CheckNoZeroAddresses(); err != nil {
+			return err
+		}
+		if chain.Eip1559DenominatorCanyon != standard.Eip1559DenominatorCanyon ||
+			chain.Eip1559Denominator != standard.Eip1559Denominator ||
+			chain.Eip1559Elasticity != standard.Eip1559Elasticity {
+			return fmt.Errorf("%w: chainId=%s", ErrNonStandardValue, chain.ID)
+		}
+		if chain.BaseFeeVaultRecipient == emptyAddress ||
+			chain.L1FeeVaultRecipient == emptyAddress ||
+			chain.SequencerFeeVaultRecipient == emptyAddress {
+			return fmt.Errorf("%w: chainId=%s", ErrFeeVaultZeroAddress, chain.ID)
+		}
+	}
+
+	return nil
+}
+
+func getStandardSuperchainRoles(l1ChainId uint64) (*SuperchainRoles, error) {
+	superCfg, err := standard.SuperchainFor(l1ChainId)
+	if err != nil {
+		return nil, fmt.Errorf("error getting superchain config: %w", err)
+	}
+
+	proxyAdmin, _ := standard.ManagerOwnerAddrFor(l1ChainId)
+	guardian, _ := standard.GuardianAddressFor(l1ChainId)
+
+	superchainRoles := &SuperchainRoles{
+		ProxyAdminOwner:       proxyAdmin,
+		ProtocolVersionsOwner: common.Address(*superCfg.Config.ProtocolVersionsAddr),
+		Guardian:              guardian,
+	}
+
+	return superchainRoles, nil
+}
+
+func (c *Intent) SetInitValues(l2ChainIds []common.Hash) error {
+	switch c.IntentConfigType {
+	case IntentConfigTypeCustom:
+		return c.setCustomValues(l2ChainIds)
+
+	case IntentConfigTypeStandard:
+		return c.setStandardValues(l2ChainIds)
+
+	case IntentConfigTypeStrict:
+		return c.setStrictValues(l2ChainIds)
+
+	case IntentConfigTypeStrictOverrides:
+		return c.setStrictValues(l2ChainIds)
+
+	case IntentConfigTypeStandardOverrides:
+		return c.setStandardValues(l2ChainIds)
+
+	case IntentConfigTypeTest:
+		return c.setTestValues(l2ChainIds)
+
+	default:
+		return fmt.Errorf("intent config type not supported")
+	}
+}
+
+// Sets all Intent fields to their zero value with the expectation that the
+// user will populate the values before running 'apply'
+func (c *Intent) setCustomValues(l2ChainIds []common.Hash) error {
+	c.L1ContractsLocator = &artifacts.Locator{Tag: "undefined"}
+	c.L2ContractsLocator = &artifacts.Locator{Tag: "undefined"}
+
+	c.SuperchainRoles = &SuperchainRoles{}
+	for _, l2ChainID := range l2ChainIds {
+		c.Chains = append(c.Chains, &ChainIntent{
+			ID: l2ChainID,
+		})
+	}
+	return nil
+}
+
+func (c *Intent) setStandardValues(l2ChainIds []common.Hash) error {
```

Comment:
All these private `set*` methods can be private initializer methods. I try and avoid putting this kind of logic on the intent - it should really just be data, with minimal logic.